posts/views.py

```diff
@@ -28,3 +30,17 @@ def post(self, request: HttpRequest, category_id: int) -> HttpResponse:
             return redirect("categories:list")
 
         return render(request, self.template_name, {"form": form, "category": category})
+
+
+class DetailsPostView(View):
+    template_name = "posts/post_detail.html"
+    template_name_2 = "posts/post_unavailable.html"
+
+    def get(self, request: HttpRequest, post_id: int) -> HttpResponse:
+        post = Post.objects.get(pk=post_id)
+        context = "This post is currently unavailable"
+
+        if post.is_active:
+            return render(request, self.template_name, {"post": post})
+
+        return render(request, self.template_name_2, {"context": context})
```

Comment:
there is specific method I remember in `django.shortcuts`. It will raise 404 error. So u can add 404.html template if u want, otherwise there will be some default page generated
```suggestion
        post = get_object_or_404(Post, pk=post_id, is_active=True)
```

Reply:
this shortcut works well. Is my variant was right?

Reply:
mostly but it is the best practice to return 404 error if object not found